crates/sui-graphql-rpc/src/types/chain_identifier.rs

```diff
@@ -39,4 +47,16 @@ impl ChainIdentifier {
             .map_err(|e| Error::Internal(format!("Failed to deserialize genesis digest: {e}")))?;
         Ok(NativeChainIdentifier::from(genesis_digest))
     }
+
+    pub(crate) async fn new(lock: ChainIdentifierLock) -> Self {
+        let w = lock.0.read().await;
+
+        Self(w.0)
+    }
```

Comment:
I would rather put a function on `ChainIdentifierLock` to return a `ChainIdentifier` than have this function here, because it introduces a circular dependency (which is fine for Rust -- until you want to split a crate up -- but confusing for humans).